packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -179,132 +177,202 @@ class _ZoomPageTransition extends StatefulWidget {
-  // curves.
-  bool get _transitionWasInterrupted {
-    bool wasInProgress = false;
-    bool isInProgress = false;
-
-    switch (_currentAnimationStatus) {
-      case AnimationStatus.completed:
-      case AnimationStatus.dismissed:
-        isInProgress = false;
-        break;
-      case AnimationStatus.forward:
-      case AnimationStatus.reverse:
-        isInProgress = true;
-        break;
-    }
-    switch (_lastAnimationStatus) {
-      case AnimationStatus.completed:
-      case AnimationStatus.dismissed:
-        wasInProgress = false;
-        break;
-      case AnimationStatus.forward:
-      case AnimationStatus.reverse:
-        wasInProgress = true;
-        break;
-    }
-    return wasInProgress && isInProgress;
-  }
+  final Animation<double> animation;
+  final Widget child;
+  final bool reverse;
 
-  @override
-  Widget build(BuildContext context) {
-    final Animation<double> _forwardScrimOpacityAnimation = widget.animation.drive(
-      _ZoomPageTransition._scrimOpacityTween
-        .chain(CurveTween(curve: const Interval(0.2075, 0.4175))));
+  static final Animatable<double> _fadeInTransition = Tween<double>(
+    begin: 0.0,
+    end: 1.00,
+  ).chain(CurveTween(curve: const Interval(0.125, 0.250)));
 
-    final Animation<double> _forwardEndScreenScaleTransition = widget.animation.drive(
-      Tween<double>(begin: 0.85, end: 1.00)
-        .chain(_ZoomPageTransition._scaleCurveSequence));
+  static final Animatable<double> _scaleDownTransition = Tween<double>(
+    begin: 1.10,
+    end: 1.00,
+  ).chain(_ZoomPageTransition._scaleCurveSequence);
 
-    final Animation<double> _forwardStartScreenScaleTransition = widget.secondaryAnimation.drive(
-      Tween<double>(begin: 1.00, end: 1.05)
-        .chain(_ZoomPageTransition._scaleCurveSequence));
+  static final Animatable<double> _scaleUpTransition = Tween<double>(
+    begin: 0.85,
+    end: 1.00,
+  ).chain(_ZoomPageTransition._scaleCurveSequence);
 
-    final Animation<double> _forwardEndScreenFadeTransition = widget.animation.drive(
-      Tween<double>(begin: 0.0, end: 1.00)
-        .chain(CurveTween(curve: const Interval(0.125, 0.250))));
+  static final Animatable<double> _scrimOpacityTween = Tween<double>(
+    begin: 0.0,
+    end: 0.60,
+  ).chain(CurveTween(curve: const Interval(0.2075, 0.4175)));
 
-    final Animation<double> _reverseEndScreenScaleTransition = widget.secondaryAnimation.drive(
-      Tween<double>(begin: 1.00, end: 1.10)
-        .chain(_ZoomPageTransition._flippedScaleCurveSequence));
+  @override
+  Widget build(BuildContext context) {
+    double opacity = 0;
+    // When the transition is running with the 'reverseBuilder' in 'DualTransitionBuilder',
+    // the status of the animation will be always [AnimationStatus.completed].
+    //
+    // To ensure the opacity only changed with forward animation, we need to combine
+    // the 'reverse' flag with the animation status.
```

Comment:
Thanks for improving the wording here, the original I left in here was really vague. Let me try to clarify this more, you can use what you think makes the most sense from it and add parts of yours that you think is helpful as well: 
```
    // The transition's scrim opacity only increases on the forward transition. In the reverse
    // transition, the opacity should always be 0.0.
    //
    // Therefore, we need to only apply the scrim opacity animation when the transition
    // is running forwards.
```

cc/ @goderbauer I think I tried `if (!reverse && animation.status == AnimationStatus.forward)` but it did not behave as I expected, but I cannot remember the exact details of the weird behavior.

Reply:
I'll take your wording, because the reason I've written which related to the `DualTransitionBuilder` seems unnecessary here.

Reply:
I looked into it again for the reason it has to be `if (!reverse && animation.status != AnimationStatus.completed)`. This allows an interrupted reversal of the forward transition to smoothly fade the scrim away.

Using `if (!reverse && animation.status == AnimationStatus.forward)` causes the scrim to instantly vanish if the animation was interrupted while it was transitioning forwards. This results in a disjointed experience, so it's likely better to have the scrim fade away rather than having it suddenly disappear.

Reply:
So should I take further action with this comment?

Reply:
I was thinking of waiting for what goderbauer thought, but I think it also makes sense to add a few sentences explaining this:
```
    // The transition's scrim opacity only increases on the forward transition. In the reverse
    // transition, the opacity should always be 0.0.
    //
    // Therefore, we need to only apply the scrim opacity animation when the transition
    // is running forwards.
    //
    // The reason that this we check that the animation's status is not `completed` instead
    // of checking that it is `forward` is that this allows the interrupted reversal of the
    // forward transition to smoothly fade the scrim away. This prevents a disjointed 
    // removal of the scrim.
```

Reply:
Then I'll hold off the commit since there are already 31 commits in this PR. But he has disappeared for a week. 🤣

Reply:
The number of commits shouldn’t matter for the PR since we squash and merge (everything shows up as a single commit).

Reply:
Okay then I'm going to mark this as resolved.